src/mixins/java/org/spongepowered/common/mixin/api/minecraft/server/level/ChunkHolderMixin_API.java

```
@@ -0,0 +1,59 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.api.minecraft.server.level;
+
+import net.minecraft.server.level.ChunkHolder;
+import net.minecraft.world.level.chunk.ImposterProtoChunk;
+import net.minecraft.world.level.chunk.LevelChunk;
+import org.spongepowered.api.ResourceKey;
+import org.spongepowered.api.event.SpongeEventFactory;
+import org.spongepowered.api.event.world.chunk.ChunkEvent;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.common.SpongeCommon;
+import org.spongepowered.common.event.ShouldFire;
+import org.spongepowered.common.event.tracking.PhaseTracker;
+import org.spongepowered.common.util.VecHelper;
+import org.spongepowered.math.vector.Vector3i;
+
+@Mixin(ChunkHolder.class)
+public abstract class ChunkHolderMixin_API {
+    @Inject(
+            method = "replaceProtoChunk(Lnet/minecraft/world/level/chunk/ImposterProtoChunk;)V",
+            at = @At("TAIL")
+    )
+    private void impl$chunkGenerated(final ImposterProtoChunk imposter, final CallbackInfo ci) {
+        if (ShouldFire.CHUNK_EVENT_GENERATED) {
```

I almost prefer a fail fast (it compiles down to the same thing anyways) like
```java
if (!ShouldFire.CHUNK_EVENT_GENERATED) {
  return;
}
// throw event
```